modules/petzku/util.moon

```diff
@@ -106,29 +107,71 @@ with lib
     .io = {
         :pathsep,
         run_cmd: (cmd, quiet) ->
-            aegisub.log 'running: %s\n', cmd unless quiet
+            aegisub.log 5, 'Running: %s\n', cmd unless quiet
 
-            local output
+            local runner_path
+            output_path = os.tmpname()
             if pathsep == '\\'
+                -- windows
                 -- command lines over 256 bytes don't get run correctly, make a temporary file as a workaround
-                tmp = aegisub.decode_path('?temp' .. pathsep .. 'tmp.bat')
-                f = io.open tmp, 'w'
-                f\write cmd
+                runner_path = aegisub.decode_path('?temp' .. pathsep .. 'petzku.bat')
+                wrapper_path = aegisub.decode_path('?temp' .. pathsep .. 'petzku-wrapper.bat')
+                exit_code_path = os.tmpname()
+                -- provided by https://sourceforge.net/projects/unxutils/
+                tee_path = "#{re.match(debug.getinfo(1).source, '@?(.*[/\\\\])')[1].str}util/tee"
+                -- create wrapper
+                f = io.open wrapper_path, 'w'
+                f\write "@echo off\n"
+                f\write "call %*\n"
+                f\write "echo %errorlevel% >\"#{exit_code_path}\"\n"
+                f\close!
+                -- create batch script
+                f = io.open runner_path, 'w'
+                f\write "@echo off\n"
+                f\write "call \"#{wrapper_path}\" #{cmd} 2>&1 | \"#{tee_path}\" \"#{output_path}\"\n"
+                f\write "set /p errorlevel=<\"#{exit_code_path}\"\n"
+                f\write "exit /b %errorlevel%\n"
                 f\close!
-
-                p = io.popen tmp
-                output = p\read '*a'
-                p\close!
-
-                os.execute 'del ' .. tmp
             else
-                -- on linux, we should be fine to just execute the command directly
-                p = io.popen cmd
-                output = p\read '*a'
-                p\close!
-
-            aegisub.log output unless quiet
-            output
+                runner_path = aegisub.decode_path('?temp' .. pathsep .. 'petzku.sh')
+                pipe_path = os.tmpname()
+                -- create shell script
+                f = io.open runner_path, 'w'
+                f\write "#!/bin/sh\n"
+                f\write "mkfifo \"#{pipe_path}\"\n"
+                f\write "tee \"#{output_path} <\"#{pipe_path}\" &\n"
+                f\write "#{cmd} >\"#{pipe_path}\" 2>&1\n"
+                f\write "exit $?\n"
+                f\close!
+                -- make the script executable
+                os.execute "chmod +x \"#{runner_path}\""
+
+            
+            status, reason, exit_code = os.execute runner_path
+            os.execute 'del ' .. runner_path
```

Comment:
should be `rm` on unix systems.

Reply:
actually, we even don't need to delete these files.